[IV-25-26] Objetivo3

oscar0310 opened the pull request:
Para completar este objetivo se ha creado un documento para justificar la elección de los gestores de versiones, dependencias y tareas junto con el lenguaje , con una explicación de como usarlos y se han añadido los archivos de configuración . Se ha creado la tarea check para comprobar la sintaxis de las entidades,

closes #24 

* [x] ¿Incluye el fichero `iv.yaml` las claves de todos los objetivos
anteriores?
* [x] ¿Se ha desarrollado el fichero para las tareas a partir de un
   issue?
* [x] ¿Ese issue está asignado a un milestone?
* [x] ¿El milestone incluye, entre sus productos mínimamente viables, la
  existencia de herramientas que permitan automatizar tareas?
* [x] ¿Hemos justificado correctamente en caso de elegir un gestor de
   dependencias el que se use como gestor de tareas?
* [x] ¿El PR, que también es un issue, está asignado al mismo milestone?
* [x] ¿La tarea `check` efectivamente comprueba la sintaxis de los fuentes
  existentes y sólo eso?
* [x] ¿Se ha comprobado que las instrucciones incluidas en el `README.md` son
completas y correctas?

oscar0310 committed:
fix #20 Añadiendo un user-journey

 Se añade un user-journey para ayudar a entender el problema

oscar0310 committed:
fix #20 Arreglando formato fichero user-journey

oscar0310 committed:
fix #20 Añadiendo un ejemplo de entrada del problema

oscar0310 committed:
fix #20 Añadiendo enlace del user-journey al README.md

oscar0310 committed:
fix #20 arreglando el enlace del user-journey

oscar0310 committed:
fix #20 arreglando formato de ejemplo del user-journey

oscar0310 committed:
fix #24 añadiendo el archivo documentacion_gestores.md

Archivo para justificar el uso de los gestores elegidos actualmente contiene la justificación del lenguaje  y gestor de versiones

oscar0310 committed:
fix #24 Añadiendo justificación gestor de dependencias a documentacion_gestores.md

oscar0310 committed:
fix #24 Añadiendo archivo de configuracin Bundler-> Gemfile

oscar0310 committed:
fix #24 Añadiendo justificación de gestores de tareas a documentacion_gestores.md

oscar0310 committed:
fix #24 Añadiendo Gemfile con rake

Se añade rake al archivo de gestión de dependencias, para que sea controlado por el gestor

oscar0310 committed:
fix #24 Añadiendo Gemfile.lock

Registra la versión de las gemas detallada una vez instaladas

oscar0310 committed:
fix #24 Añadiendo .ruby-version

Contiene la versión del proyecto

oscar0310 committed:
fix #24 Añadiendo Rubocop a las dependencias

Linter que controlará la sintaxis de nuestro repositorio

oscar0310 committed:
fix #24 añadiendo configuración de Rubocop

Solo analiza entidades

oscar0310 committed:
fix #24 añadiendo Rakefile con la tarea check

oscar0310 committed:
fix #24 añadiendo la clave automatizar con las subclaves a iv.yaml

oscar0310 committed:
fix #24 actualizando README.md

oscar0310 committed:
fix #24 Arreglando el comando de orden

GabrielFranciscoSM commented on docs/documentacion_gestores.md:
```diff
@@ -0,0 +1,110 @@
+## GESTORES DE TAREAS
+Un **gestor de tareas** es el encargado de ralizar estas tareas tan repetitivas, estas operaciones se definen en un fichero, permitiendo que se ejecuten con un simple comando.
+
+En ruby existen varias opciones en gestores de tareas, **como son Rake, Thor y Make**, a la hora de elegir me he centrado primero en el estandar, la curva de aprendizaje y como interactuan con el proyecto. En cuanto a la curva de aprendizaje, **Make** sería el más tedioso porque tendríamos que usar el lenguaje de la shell de linux, mientras que **Thor y Rake** usan ruby ganando puntos a favor. A  ...
```
¿la curva de aprendizaje y como interactuan con el proyecto son requisitos medibles y objetivos?

GabrielFranciscoSM reviewed (commented):
Hola, te he comentado algunas cosillas.

oscar0310 committed:
#24 Arreglando la justificación del gestor de versiones

oscar0310 committed:
#24 ELiminando linter

 No es necesario el linter para chequear la sintaxis

oscar0310 committed:
#24 Arreglando la justificación del gestor de dependencias

oscar0310 committed:
#24 Arreglando la justificación del gestor de Tareas

oscar0310 committed:
#24 Arreglando la justificacion de gestor de versiones

Añado rv

oscar0310 committed:
#24 Añadiendo imagen para justificación de rv

oscar0310 committed:
#24 Añadiendo la orden para comprobar la sintaxis

Añadimos la tarea al rakefile para comprobar la sintaxis de todos los .rb y la orden en el README para ejecutarlo

GabrielFranciscoSM commented on docs/documentacion_gestores.md:
```diff
@@ -0,0 +1,109 @@
+## GESTORES DE TAREAS
+Un **gestor de tareas** es el encargado de ralizar estas tareas tan repetitivas, estas operaciones se definen en un fichero, permitiendo que se ejecuten con un simple comando.
+
+En ruby existen varias opciones en gestores de tareas, **como son Rake, Thor y Make**, a la hora de elegir me he centrado primero en el estandar y deuda técnica. En cuanto a la deuda técnica, **Make** usa el lenguaje de la shell de linux teniendo que obligar a quien trabajae en el proyecto a conocerla, mientras que **Thor y Rake** usan ruby. En cuanto al estandar [Rake](https://www.ruby-toolbox.c ...
```
Igual

GabrielFranciscoSM reviewed (commented):
Creo que está bastante mejor, pero te he comentado sobre usar deuda técnica como criterio. Intenta dejar los criterios claros, que sean objetivos y medibles. Si lo consigues ya lo tienes.

oscar0310 committed:
#24 Especificando los criterios

Se cambia la deuda técnica como criterio por criterios más específicos y medibles

GabrielFranciscoSM commented on docs/documentacion_gestores.md:
```diff
@@ -22,12 +22,12 @@ El gestor de versiones nos permite tener varias versiones de un mismo lenguaje e
 Para trabajar con **ruby** a la hora de elegir un gestor de versiones me he fijado en : **rbenv** que es el estandar moderno, **RVM** que es el clásico y **rv** una nueva herramienta todo en uno aunque actualmente solo gestiona versiones. Vamos a enumerar pros y contras para decantarnos por uno fijandonos en lo siguientes factores:
     - En la velocidad.
-    - Deuda Técnica.
+    - La herramienta que menor número de problemas pueda provocar en el futuro.
```
Esta mejor creo, porque ya es medible (número de problemas). Pero ¿Es algo objetivo? Es más ¿Podemos saber a ciencia cierta que problemas pueda provocar?

GabrielFranciscoSM reviewed (commented):
Entiendo por dónde quieres ir, pero creo que deberías de darle una vuelta a cómo lo expresas.

gosema commented on docs/documentacion_gestores.md:
```diff
@@ -0,0 +1,109 @@
+
+A la hora de elegir el gestor de dependencias en **ruby**, No hay una gran cantidad donde elegir , podemos hablar del estandar que es [Bundler](https://bundler.io/), [Gel](https://github.com/gel-rb/gel) y podemos mencionar a [rv](https://github.com/spinel-coop/rv), que actualmente en la fecha de realización de este proyecto, es solo un gestor de versiones pero en un futuro promete ser una herremi ...
+
+Como criterio de elección nos vamos a centrar en la herramienta provoque menos problemas, Gel es una herramienta bastante desactualizada, la versión que te instala gem es la 0.3 (la última estable) la cual usa el métododo **use_gemdeps** que se eliminó a partir de la versión 3 de Ruby por tanto esto nos imposibilita usarlo, pese a ser más rápido que Bundler como se describe en su repositorio, por ...
```
No entiendo del todo este criterio, usas la que menos problemas te de pero ¿en que? Instalando, sus versiones, su documentación.... ¿Qué problemas exactamente son los que usas como criterio? ¿Es tu criterio que tenga el método use_gemdeps? Como te ha dicho mi compañero, ¿esto es objetivo?

gosema commented on docs/documentacion_gestores.md:
```diff
@@ -0,0 +1,109 @@
+## GESTORES DE TAREAS
+Un **gestor de tareas** es el encargado de ralizar estas tareas tan repetitivas, estas operaciones se definen en un fichero, permitiendo que se ejecuten con un simple comando.
+
+En ruby existen varias opciones en gestores de tareas, **como son Rake, Thor y Make**, a la hora de elegir me he centrado primero en el estandar y en la perdida de tiempo aprendiendo aprendiendo un nuevo lenguaje para usarla. En cuanto a la perdida de tiempo, **Make** usa el lenguaje de la shell de linux teniendo que obligar a quien trabaja en el proyecto a conocerla, mientras que **Thor y Rake** ...
```
¿Crees que es un criterio válido para tu proyecto la dificultad del lenguaje? ¿Esto es objetivo?¿Si trabajas con un desarrollador experto en lenguaje shell compartirá este criterio con el tuyo?
Y sobre el estándar ¿porque lo consideras un criterio y que objetividad aporta? ¿Que se use más significa que sea mejor?

gosema reviewed (commented):
Buenas tardes @oscar0310 . Creo que has hecho una gran labor investigando los gestores posibles en este lenguaje, pero creo que todavía no estás utilizando criterios totalmente objetivos. Deberías darle una vuelta a esta idea y buscar unos más válidos

oscar0310 committed:
#24 Actualizando la elección del gestor de versiones

Estableciendo como único criterio la velocidad

oscar0310 committed:
#24 Arreglando justificación gestor de dependencias

Establecemos como criterio único la última vez que se actualizó

oscar0310 committed:
#24 Arreglando la justificación del gestor de tareas

oscar0310 committed:
#24 Eliminando todo lo innnecesario del Objetivo3

He eliminado todo lo que no tenga que ver con el objetivo 3 como JJ me ha recomendado en clase

oscar0310 committed:
Merge branch 'main' into Objetivo3

oscar0310 commented:
Buenas @gosema y @GabrielFranciscoSM  he actualizado el pull request eliminando todo lo innecesario, en cuanto a las justificaciones me gustaría aclarar que lo he estado comentando con @JJ he establecido como criterio que la herramienta sea del ecosistema de ruby y hay dos gestores de tareas  que lo cumplen, ahí es donde puedo elegir ya la estándar o popular. Si pudierais darme vuestra opinión de como lo veis.

GabrielFranciscoSM reviewed (approved):
Veo que has reducido bastante la argumentación, dejando sólo los criterios más fundamentales. Con esto creo que es una elección justificada y suficiente.

oscar0310 commented:
@gosema cuando puedas, le echas un vistazo porfa.

gosema reviewed (approved):
Siento no contestar antes @oscar0310 .
Esta muy bien tu proyecto, has entendido los criterios objetivos y has eliminado el ruido de la documentación.

oscar0310 commented:
@JJ listo para revisión.

JJ commented on docs/documentacion_gestores.md:
```diff
@@ -0,0 +1,38 @@
+## GESTORES DE TAREAS
+Un **gestor de tareas** es el encargado de ralizar estas tareas tan repetitivas, estas operaciones se definen en un fichero, permitiendo que se ejecuten con un simple comando.
+
+En ruby existen varias opciones en gestores de tareas, **como son Rake, Bake, Task, Just y Make**, a la hora de elegir me he centrado en una herramienta que esté integrada en el ecosistema ruby como son Rake y Bake, ambas cumplen el criterio pero me he decantado por **Rake** por ser el estandar.
```
No has puesto ningún criterio de antemano. Aunque rake sea el estándar, lo será por alguna razón; también make es estándar en Linux y sin embargo no dices nada.
También deberías poner enlaces como lo has puesto arriba. Te has currado bastante esa parte, pero no esta.

JJ reviewed (commented):
Aparte de lo dicho, se cuidadoso con la tipografía: no se usa espacio antes de coma, ni después de paréntesis.
También sería conveniente que si hay un estándar para especificar las dependencias, lo mencionaras, y a los gestores de dependencias en función de si lo usan o no, lo que te permitiría descartar alguno de antemano.
Si ese estándar para especificar dependencias permite también ejecutar tareas o especificar tareas, deberías incluirlo también.
Finalmente, no especificas si se trata de un estándar de facto en ninguno de los dos casos, o son herramientas que se instalan junto con el lenguaje. Sería conveniente saberlo.

oscar0310 committed:
#24 ELiminando errores de tipografía

oscar0310 committed:
#24 actualizando la justificación del gestor de dependencias

 Explico el estandar para especificar las dependencias, añado si lo usan o no los gestores, la forma de instalación y cual es el estandar de facto

oscar0310 committed:
#24 actualizando la justificación del gestor de tareas

Defino más en detalle las distinas opciones añadiendo links con información , describiendo cual es el estandar de facto y los criterios elegidos para la elección del gestor

oscar0310 committed:
#24 Arreglando erratas ortográficas

oscar0310 commented:
Buenas @GabrielFranciscoSM y @gosema , he implementado algunos cambios tras la revisión de @JJ, podríais mirarlos a ver que os parecen.

GabrielFranciscoSM commented on docs/documentacion_gestores.md:
```diff
@@ -28,11 +28,13 @@ Por tanto nos quedamos con el estandar de facto Bundler.
+En ruby existen varias opciones en gestores de tareas, podemos hablar de herramientas integradas en el escositema Ruby como son [Rake](https://ruby.github.io/rake/) siendo una gema de ruby, la cual podríamos instalar con Bundler y es actualmente el estandar de [facto](https://www.ruby-toolbox.com/projects/rake) al ser la herramienta más dominante y usada en cuanto a gestores de tareas en ruby, [B ...
 
 
+A la hora de elegir he establecido como criterio que la herramienta que vamos a usar esté integrada en el ecosistema ruby y dentro de esas herramientas nos quedaremos con el estandar de facto. Esto descartaría tanto a Just, Task y Make que son herramientas externas. Y nos dejaría a Bake y Rake ambas cumplen la condición de estar integradas en el ecosistema ruby, siendo gems del propio lenguaje, p ...
```
¿Crees que es adecuado declarar los criterios de elección después de describir los gestores de tareas según estos criterios?

GabrielFranciscoSM commented on docs/documentacion_gestores.md:
```diff
@@ -0,0 +1,47 @@
+- Instalar las gemas desde Git o desde un path local.
+- Especificar condiciones para instalar las gemas.
+
+A la hora de elegir el gestor de dependencias en **ruby**, No hay una gran cantidad donde elegir, podemos hablar de [Bundler](https://bundler.io/), el cual viene instalado por defecto en las versiones modernas de **ruby** siendo el [estandar de facto](https://www.ruby-toolbox.com/categories/dependency_management) y usa el estandar para especificar las dependencias. [Gel](https://github.com/gel-rb ...
```
```suggestion
A la hora de elegir el gestor de dependencias en **ruby**, no hay una gran cantidad donde elegir, podemos hablar de [Bundler](https://bundler.io/), el cual viene instalado por defecto en las versiones modernas de **ruby** siendo el [estandar de facto](https://www.ruby-toolbox.com/categories/dependency_management) y usa el estandar para especificar las dependencias. [Gel](https://github.com/gel-rb/gel) que es una gem de ruby, habría que instalarlo con el gestor de paquetes de **ruby** y también usa el estandar para especificar las dependencias. Podemos mencionar a [rv](https://github.com/spinel-coop/rv), que actualmente en la fecha de realización de este proyecto, es solo un gestor de versiones pero en un futuro promete ser una herramienta todo en uno, muy a tener en cuenta, habría que instalarlo desde su repositorio y se desconoce si usará el estandar para especificar las herramientas.
```

GabrielFranciscoSM reviewed (approved):
Te lo acepto, pero te he hecho un comentario sobre la estructura lógica. El contenido lo veo bien.

oscar0310 committed:
#24 Error ortográfico

Co-authored-by: GabrielFranciscoSM <100140435+GabrielFranciscoSM@users.noreply.github.com>

gosema reviewed (approved):
Tienes mi visto bueno, concuerdo con Gabriel-

oscar0310 committed:
#24 alterando el orden de la justificación de el gestor de tareas

 Establezco primero los criterioss, luego enumero las herramientas y por último elijo

oscar0310 commented:
Siguiendo vuestras recomendaciones he ordenado lógicamente la justificación del gestor de tareas. Muchas gracias por el feedback @GabrielFranciscoSM y @gosema .

oscar0310 commented:
@JJ  listo para revisión.

JJ commented on docs/documentacion_gestores.md:
```diff
@@ -0,0 +1,49 @@
+## GESTORES DE TAREAS
+Un **gestor de tareas** es el encargado de realizar estas tareas tan repetitivas, estas operaciones se definen en un fichero, permitiendo que se ejecuten con un simple comando.
+
+Como criterio para elegir establecemos que la herramienta que vamos a usar esté integrada en el ecosistema ruby y dentro de esas herramientas nos quedaremos con el estandar de facto.
```
Imagino que con esto quieres excluir otras como make o just.

oscar0310 replied:
Siii

JJ commented on docs/documentacion_gestores.md:
```diff
@@ -0,0 +1,49 @@
+
+Como criterio para elegir establecemos que la herramienta que vamos a usar esté integrada en el ecosistema ruby y dentro de esas herramientas nos quedaremos con el estandar de facto.
+
+En ruby existen varias opciones en gestores de tareas, podemos hablar de herramientas integradas en el ecositema Ruby como son [Rake](https://ruby.github.io/rake/) siendo una gema de ruby, la cual podríamos instalar con Bundler y es actualmente el estandar de [facto](https://www.ruby-toolbox.com/projects/rake) al ser la herramienta más dominante y usada en cuanto a gestores de tareas en ruby, [Ba ...
```
```suggestion
En ruby existen varias opciones en gestores de tareas, podemos hablar de herramientas integradas en el ecosistema Ruby como son [Rake](https://ruby.github.io/rake/) siendo una gema de ruby, la cual podríamos instalar con Bundler y es actualmente el [estandar de facto](https://www.ruby-toolbox.com/projects/rake) al ser la herramienta más dominante y usada en cuanto a gestores de tareas en ruby, [Bake](https://ioquatix.github.io/bake/) es otra gema la cual puede ser intalada tambien con Bundler. Herramientas externas al ecosistema ruby como son [Task](https://taskfile.dev/) y [Just](https://github.com/casey/just) que son herramientas  multiplataforma insparada en Make y por último [Make](https://www.gnu.org/software/make/manual/make.html) el estandar actual de linux viniendo instalada por defecto en todas las distribuciones de linux. 
```

JJ reviewed (approved):
👍

oscar0310 committed:
#24 Arreglando error ortográfico

Co-authored-by: Juan Julián Merelo Guervós <jjmerelo@gmail.com>

JJ commented:
Estaba ya aprobado, puedes fusionar cuando quieras.
